snuba/settings/__init__.py

```diff
@@ -144,6 +144,11 @@
 
 # Processor/Writer Options
 
+# Mapping of (logical topic names, slice id) pairs to physical topic names
+# This is for Kafka topics that are associated
+# with multiple slices
+SLICED_KAFKA_TOPICS: Mapping[Tuple[str, int], str] = {}
```

Comment:
IS this supposed to be equivalent to `KAFKA_TOPIC_MAP` but for slices? If so please place it directly below the other variable and describe the interaction between the two settings. I also kind of wonder if we really need to support this right now. Why not just have every slice use the same topic names?

Reply:
Separate physical topic names were raised by @fpacifici as a hard requirement, no compromise possible. Happy to consider the possibility of same topic names with different broker connections if my understanding was wrong

Reply:
That's fine, just a though. Please make the variable names mirror the original ones so the relationship is clearer though.

Reply:
I think because self-hosted and single-tenant have using a single Kafka broker as a requirement (though to your point, it doesn't seem highly likely to me that they'll need slicing and not be able to bring in a second broker)

Reply:
>  As for no compromise possible, curious why.

Compromise is always possible. Though here we would be compromising on the ability to consolidate Kafka clusters.
If we have two topics in the fleet with the same name, they will have to be on two separate brokers forever. 
For all practical purpose, Kafka is one of the most scalable components we have, likely we could have shared the broker between two slices. Sharing the name would prevent us from doing so.

So, taking a step back, allowing name conflicts between topics means the application imposing an operational constraint for no real reason. Keeping the names separate allows us to be flexible when we deploy in multiple environments as we can consolidate or split topic as we want.